hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactingMemStore.java

```diff
@@ -467,64 +485,69 @@ private void doCompact() {
       } catch (Exception e) {
         LOG.debug("Interrupting the MemStore in-memory compaction for store " + getFamilyName());
         Thread.currentThread().interrupt();
-        return;
       } finally {
         releaseResources();
         inMemoryFlushInProgress.set(false);
       }
-
     }
 
-    /**
-     * Creates the scanner for compacting the pipeline.
-     *
-     * @return the scanner
+    /**----------------------------------------------------------------------
+     * The compaction is the creation of the relevant ImmutableSegment based on
+     * the Compactor Itertor
      */
-    private StoreScanner createScanner(Store store) throws IOException {
-
-      Scan scan = new Scan();
-      scan.setMaxVersions();  //Get all available versions
+    private ImmutableSegment compact(int numOfCells)
+        throws IOException {
 
-      StoreScanner internalScanner =
-          new StoreScanner(store, store.getScanInfo(), scan, Collections.singletonList(scanner),
-              ScanType.COMPACT_RETAIN_DELETES, smallestReadPoint, HConstants.OLDEST_TIMESTAMP);
+      ImmutableSegment result = null;
+      MemStoreCompactorIterator iterator =
+          new MemStoreCompactorIterator(versionedList.getStoreSegments(), getComparator(),
+              compactionKVMax, store);
+      try {
+        switch (type) {
```

Comment:
It's not compaction logic, just creation of the segment. Normally the factory would receive only the constant